hack/install-vm.sh

```diff
@@ -0,0 +1,407 @@
+EOF
+    systemctl daemon-reload
+    systemctl enable "$ANTREA_AGENT"
+    echo "Starting ${ANTREA_AGENT} service"
+    systemctl start "$ANTREA_AGENT"
+    systemctl status "$ANTREA_AGENT"
+}
+​​
+# Generic function to execute command with timeout
+execute_command() {
+    cmd="$1"
+    timeout="$2"
+    echo "Executing $cmd with timeout $timeout at $(date)"
+    timeout --preserve-status --foreground $timeout $cmd
+    ret=$?
+    if [ $ret -ne 0 ]; then
+	      echoerr "$cmd failed with status $ret at $(date)"
+        echoerr "Exiting!! please check internet connectivity and re-run install script"
+        exit 2
+    fi
+}
+​
+# Install docker as container runtime
+install_container_runtime() {
+    os=`grep -Po '^ID=\K.*' /etc/os-release | sed -e 's/^"//' -e 's/"$//'`
+    echo "Installing container runtime on $os"
+    case "$os" in
+        ubuntu)
+	    # Uninstall older versions of Docker(called docker, docker.io, or docker-engine)
+            timeout --preserve-status --foreground 1m apt-get remove docker docker-engine docker.io containerd runc
+            timeout --preserve-status --foreground 1m apt-get update
+	    cmd="apt-get update"
+	    execute_command "$cmd" 1m
+	    cmd="apt-get install -y ca-certificates curl gnupg lsb-release"
+	    execute_command "$cmd" 1m
+	    # Add docker's offical GPG key
+	    rm -rf /usr/share/keyrings/docker-archive-keyring.gpg
+	    timeout --preserve-status --foreground 10s curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
+	    # Setup a stable repository
+	    echo \
+            "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu \
+            $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+​
+	    cmd="apt-get update"
+	    execute_command "$cmd" 1m
+	    cmd="apt-get install -y docker-ce docker-ce-cli containerd.io"
+	    execute_command "$cmd" 2m
+	    ;;
+    esac
+}
+​
+# Install all pre-requisite packages needed for install
+install_docker_requisite_packages() {
+    set +eo pipefail
+    if ! command -v curl &> /dev/null; then
+        install_curl
+    fi
+    # Install docker, docker-compose only when running services in containers
+    if [ "$INSTALL_DOCKER" = true ]; then
+        docker version
+        if [[ "$?" -ne 0 ]]; then
+            echo "Docker is not installed"
+            install_container_runtime
+        fi
+        # Install docker-compose
+        docker-compose version
+        if [ "$?" -ne 0 ]; then
+            timeout --preserve-status --foreground 5s curl -L "https://github.com/docker/compose/releases/download/1.29.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+            if [ "$?" -eq 0 ]; then
+                echo "docker-compose is downloaded successfully"
+                chmod +x /usr/local/bin/docker-compose
```

Comment:
/usr/local/bin/docker-compose
Is this path the preferred/default path for docker compose?

use variables for the paths, as it will be easier to maintain going forward.